src/app/cmd/flatten_layers.cpp

```diff
@@ -66,8 +73,52 @@ void FlattenLayers::onExecute()
   if (list.empty())
     return;                     // Do nothing
 
+  // Extend the drawable area beyond the sprite bounds
+  // when this option is enabled
+  ImageSpec spec = sprite->spec();
+  int area_x = 0;
+  int area_y = 0;
+  if ((m_options & Options::ExtendCanvas) != 0) {
+    // Get initial dimentions of draw area
+    int mX = spec.width();
+    int mY = spec.height();
+
+    // Check those bounds against every frame of every layer that
+    // will be merged
+    for (frame_t frame(0); frame<sprite->totalFrames(); ++frame) {
+      for (Layer* layer : layers) {
+        Cel* cel = layer->cel(frame);
+        if (!cel)
+          continue;
+
+        // Is this cel outside of the current dimentions?
+        const gfx::Rect bounds = cel->bounds();
+        const int x = (bounds.x < 0 ? bounds.w+spec.width()+abs(bounds.x):
+                                      bounds.w+bounds.x);
+        const int y = (bounds.y < 0 ? bounds.h+spec.height()+abs(bounds.y):
+                                      bounds.h+bounds.y);
+
+        if (x > mX)
+          mX = x;
+
+        if (y > mY)
+          mY = y;
+      }
+    }
```

Comment:
Probably something we can use to simplify this is an union of all `cel->bounds()`, something like:

```c++
    gfx::Rect extends;
    for (frame_t frame(0); frame<sprite->totalFrames(); ++frame) {
      for (Layer* layer : layers) {
        Cel* cel = layer->cel(frame);
        if (cel)
          extends |= cel->bounds();
      }
    }
```
And use `extends.origin()` to replace `mX` and `mY` (we could even avoid creating a whole canvas image if it's not required for the merge down).

Reply:
I like this, it's much simpler logic, and can indeed be used to make the temporary image only as big as necessary rather than always enlarging it. I'll try to think of a more appropriate name for the `ExtendCanvas` option, so that it better reflects this more dynamic behavior.